File: android/app/src/main/java/com/adqmobile/socialnetwork/data/remote/UserService.kt
package com.adqmobile.socialnetwork.data.remote

import com.adqmobile.socialnetwork.domain.user.Login
import com.adqmobile.socialnetwork.domain.user.User
import retrofit2.Response
import retrofit2.http.Body
import retrofit2.http.POST

interface UserService {

    @POST("login")
    suspend fun postLogin(@Body login: Login): User
}

File: react-native/data/UserService.ts
import User from "../domain/user/User";
import Login from "../domain/user/Login";
import Axios from "axios";
import Service from "./Service";

export default class UserService extends Service {
    async login(login: Login): Promise<User> {
        const user = await Axios.post(`${this.HOST}/login`, login);
        return user.data;
    }
}